stacker/providers/session_cache.py

```diff
@@ -0,0 +1,64 @@
+import json
+import os
+import logging
+import botocore
+
+
+def get_session(region):
+    session = botocore.session.get_session()
+    session.set_config_variable('region',  region)
+    c = session.get_component('credential_provider')
+    provider = c.get_provider('assume-role')
+    provider.cache = CredentialCache()
+    return session
+
+
+class CredentialCache(object):
+
+    """JSON file cache.
+    This provides a dict like interface that stores JSON serializable
+    objects.
+    The objects are serialized to JSON and stored in a file.  These
+    values can be retrieved at a later time.
+    """
+
+    CACHE_DIR = os.path.expanduser(
+        os.path.join('~', '.stacker', 'cache'))
+
+    def __init__(self, working_dir=CACHE_DIR):
+        self._working_dir = working_dir
+
+    def __contains__(self, cache_key):
+        print("Called Contains")
```

Comment:
I'm guessing these are just debug statements, but they should be pulled out.  Might be worth replacing them with logging statements at the debug level though.